funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_worker.py

```diff
@@ -92,6 +94,11 @@ def __init__(self, worker_id, address, port, logdir, debug=False, worker_type='R
         logger.info('Trying to connect to : tcp://{}:{}'.format(self.address, self.port))
         self.task_socket.connect('tcp://{}:{}'.format(self.address, self.port))
         self.poller.register(self.task_socket, zmq.POLLIN)
+        signal.signal(signal.SIGTERM, self.handler)
+
+    def handler(self, signum, frame):
```

Comment:
without this handler, the process would exit immediately; with this handler, there's a log message and a TaskCancelled error raised from wherever the process is, which can be caught be anything expecting an exception - so a sigterm turns into some exception handling, rather than immediate exit?